ratis-netty/src/main/java/org/apache/ratis/netty/client/NettyClientStreamRpc.java

```diff
@@ -158,9 +196,18 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
         clientInvocationId = ClientInvocationId.valueOf(reply.getClientId(), reply.getStreamId());
         final ReplyQueue queue = replies.get(clientInvocationId);
         if (queue != null) {
-          final CompletableFuture<DataStreamReply> f = queue.poll();
+          final DataStreamRequestEntry f = queue.poll();
           if (f != null) {
-            f.complete(reply);
+            f.getReplyFuture().complete(reply);
+
+            if (!reply.isSuccess() && queue.size() > 0) {
+              replies.remove(clientInvocationId).forEach(cf -> {
+                if (!cf.getReplyFuture().isDone()) {
+                  cf.getReplyFuture().complete(cf.getErrorDataStreamReply());
+                }
+              });
+            }
```

Comment:
Just complete the futures exceptionally and don't build a reply.
```
            if (!reply.isSuccess() && queue.size() > 0) {
              final IllegalStateException e = new IllegalStateException(
                  this + ": an earlier request failed with " + reply);
              queue.forEach(future -> future.completeExceptionally(e));
            }
```

Reply:
@szetszwo  Has been modified, please take a look.